tests/components/lidarr/test_sensor.py

```diff
@@ -1,32 +1,90 @@
+from datetime import timedelta
+from unittest.mock import patch
+
+from aiopyarr.exceptions import ArrConnectionException
+import pytest
+
+from homeassistant.components.sensor import (
+    ATTR_STATE_CLASS,
+    SensorDeviceClass,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_UNIT_OF_MEASUREMENT,
+    STATE_UNAVAILABLE,
+)
 from homeassistant.core import HomeAssistant
+import homeassistant.util.dt as dt_util
+
+from . import setup_integration
 
-from .conftest import ComponentSetup
+from tests.common import async_fire_time_changed
+from tests.test_util.aiohttp import AiohttpClientMocker
 
 
+@pytest.mark.freeze_time("2021-12-03 00:00:00+00:00")
+
 async def test_sensors(
     hass: HomeAssistant,
-    setup_integration: ComponentSetup,
+    aioclient_mock: AiohttpClientMocker,
     entity_registry_enabled_by_default: None,
-    connection,
+    windows: bool,
+    single: bool,
+    root_folder: str,
 ) -> None:
     """Test for successfully setting up the Lidarr platform."""
-    await setup_integration()
+    await setup_integration(hass, aioclient_mock, windows=windows, single_return=single)
 
-    state = hass.states.get("sensor.mock_title_disk_space")
-    assert state.state == "0.93"
+    state = hass.states.get(f"sensor.mock_title_disk_space_{root_folder}")
+    assert state.state == "263.10"
     assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == "GB"
-    state = hass.states.get("sensor.mock_title_queue")
-    assert state.state == "2"
-    assert state.attributes.get("string") == "stopped"
-    assert state.attributes.get("string2") == "downloading"
-    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == "Albums"
-    assert state.attributes.get(CONF_STATE_CLASS) == SensorStateClass.TOTAL
     state = hass.states.get("sensor.mock_title_wanted")
     assert state.state == "1"
-    assert state.attributes.get("test") == "test"
     assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == "Albums"
-    assert state.attributes.get(CONF_STATE_CLASS) == SensorStateClass.TOTAL
+    state = hass.states.get("sensor.mock_title_start_time")
+    assert state.state == "2020-09-01T23:50:20+00:00"
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.TIMESTAMP
+    state = hass.states.get("sensor.mock_title_queue")
+    assert state.state == "2"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) == "Albums"
+    assert state.attributes.get(ATTR_STATE_CLASS) is SensorStateClass.TOTAL
+
+
+@pytest.mark.freeze_time("2021-12-03 00:00:00+00:00")
+async def test_windows(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test for successfully setting up the Lidarr platform on Windows."""
+    await setup_integration(hass, aioclient_mock, windows=True)
+
+    state = hass.states.get("sensor.mock_title_disk_space_tv")
+    assert state.state == "263.10"
```

Comment:
Is there OS specific code in the setup? If so, it might be good to add a comment explaining why a Windows test is necessary.